lib/utils/visUtil.hpp

```diff
@@ -261,6 +261,14 @@ inline double ts_to_double(const timespec& ts) {
     return (ts.tv_sec + 1e-9 * ts.tv_nsec);
 }
 
+/**
+ * @brief Converts a timespec type into UNIX time in milliseconds as a uint64_t.
+ * @param  ts Time as timespec.
+ * @return    Time in milliseconds as a uint64_t.
+ */
+constexpr uint64_t timespec_to_milliseconds(const timespec& ts) {
```

Comment:
It's a bit odd to use milliseconds, our base unit of time in all our measurements is nano seconds.   Generally we timestamp data to resolution of -/+ 10 nanoseconds.   This is dropping a lot of precision and risks introducing rounding errors.

Reply:
Well, since the timescale for the arriving data window timeout is in milliseconds, it is reasonable to avoid extra divisions for converting from nano- to milliseconds and having such a helper function.